Foundation/include/Poco/Dynamic/VarHolder.h

```diff
@@ -787,6 +823,287 @@ class VarHolderImpl: public VarHolder
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int32& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int64& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt8& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt16& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt32& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt64& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+#ifdef POCO_INT64_IS_LONG
+
+	void convert(long long& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(unsigned long long& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+#endif
+
+	void convert(bool& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			val = (std::underlying_type_t<T>(_val) != 0);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(float& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			val = static_cast<float>(_val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(double& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			val = static_cast<double>(_val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(char& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			val = static_cast<char>(_val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(std::string& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			val = NumberFormatter::format(std::underlying_type_t<T>(_val));
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Poco::UTF16String& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			std::string str = NumberFormatter::format(std::underlying_type_t<T>(_val));
+			Poco::UnicodeConverter::convert(str, val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	bool isArray() const
```

Comment:
I think that partial specialisation could also simplify all of the is* member functions.

Reply:
I'm not sure how would this work, consolidating all functions into one `is()` and having an enum designator? that would break a lot of code

Reply:
After taking another look, my first suggestion does not make much sense.